src/vegur_client.erl

```diff
@@ -154,12 +180,21 @@ raw_request(Data, Client=#client{
             Err
     end.
 
+%% @doc equivalent to `request(Method, URL, [], <<>>, Client)'.
 request(Method, URL, Client) ->
     request(Method, URL, [], <<>>, Client).
 
+%% @doc equivalent to `request(Method, URL, Headers, <<>>, Client)'.
 request(Method, URL, Headers, Client) ->
     request(Method, URL, Headers, <<>>, Client).
 
+%% @doc Sends an HTTP request by formatting headers and attaching the body with
+%% them. Only expects clients which are not connected, or clients which are
+%% connected following a keepalive request sent with this function.
+%% Do not pass in a pre-connected client here as you risk getting connection
+%% leaks.
+-spec request(iodata(), binary(), [{binary(), binary()}], iodata(), client()) ->
```

Comment:
in cases where there's a lot of `binary()`, I find it more readable to give the variables names and use a `when` block

may just be me though